Confus/Confus/Flag.cpp

```diff
@@ -109,71 +117,100 @@ namespace Confus {
         particleAffector->drop();
     }
 
-    irr::video::SColor Flag::getColor() {
+    irr::video::SColor Flag::getColor() 
+    {
         switch(*m_TeamIdentifier)
         {
-        case ETeamIdentifier::TEAM_BLUE:
+        case ETeamIdentifier::TeamBlue:
             return { 255, 0, 0, 255 };
-        case ETeamIdentifier::TEAM_RED:
+        case ETeamIdentifier::TeamRed:
             return { 255, 255, 0, 0 };
         default:
             return { 255, 255, 255, 255 };
         }
     }
 
 	//This class handles what to do on collision
-	void Flag::captureFlag(Player* a_PlayerObject) {
+	void Flag::captureFlag(Player* a_PlayerObject) 
+    {
 		//Somebody is already carrying the flag
-		if (*m_FlagStatus == EFlagEnum::FLAG_TAKEN)
+		if (*m_FlagStatus == EFlagEnum::FlagTaken)
 			return;
 
-		if (a_PlayerObject->TeamIdentifier != *m_TeamIdentifier) {
-			//Capture Flag
-			m_FlagNode->setParent(a_PlayerObject->PlayerNode);
+		if (*a_PlayerObject->TeamIdentifier != *m_TeamIdentifier && *a_PlayerObject->CarryingFlag == EFlagEnum::None) 
+        {
+            MyLogger->log("Capturing flag.");
+			//Capture Flag if player has no flag
+            m_FlagNode->setParent(a_PlayerObject->PlayerNode);            
+            m_FlagStatus = new EFlagEnum(EFlagEnum::FlagTaken);
```

Comment:
This leaks every time you are capturing a flag with this player object, since you aren't deleting the old flag status